yarn-project/aztec-rpc/src/note_processor/note_processor.ts

```diff
@@ -109,6 +109,10 @@ export class NoteProcessor {
           indexOfTxInABlock * MAX_NEW_COMMITMENTS_PER_TX,
           (indexOfTxInABlock + 1) * MAX_NEW_COMMITMENTS_PER_TX,
         );
+        const newNullifiers = block.newNullifiers.slice(
```

Comment:
for my understanding - why did we move this earlier?

Reply:
We were fetching the `newNullifiers` from the L2 block inside the decryption logic. This means for each successfully decrypted note, we re-do the slicing on `block.newNullifiers` to get the same value over and over again. I moved it outside to fetch the `newNullifiers` for a transaction just once.